models/weecology-esss.R

```diff
@@ -0,0 +1,155 @@
+                        LowerPI = etsmf$lower[,
+                                      which(etsmf$level == CI_level*100)], 
+                        UpperPI = etsmf$upper[,
+                                      which(etsmf$level == CI_level*100)])
+       fdt[sapply(fdt, is.ts)] <- lapply(fdt[sapply(fdt, is.ts)], unclass)
+  
+       # Include columns describing the data used in the forecast
+
+         fdt$fit_start_newmoon <- min(abundances$newmoonnumber)
+         fdt$fit_end_newmoon <- max(abundances$newmoonnumber)
+         fdt$initial_newmoon <- max(abundances$newmoonnumber)
+  
+    # prep the aic data tabe
+      
+      aic <- data.frame(date = as.Date(forecast_date), 
+                        currency = 'abundance', 
+                        model = 'Weecology-ESSS', 
+                        level = level, species = 'total', 
+                        aic = as.numeric(etsmf$model$aic), 
+                        fit_start_newmoon = min(abundances$newmoonnumber),
+                        fit_end_newmoon = max(abundances$newmoonnumber), 
+                        initial_newmoon = max(abundances$newmoonnumber))
+
+    # return the output
+
+      return(list(fdt, aic))
+  }
+
+
+
+
+# Run model on all plots and just controls
+
+  # Get data
+
+    all <- read.csv("data/rodent_all.csv")
+    controls <- read.csv("data/rodent_controls.csv")
+    model_metadata <- yaml.load_file("data/model_metadata.yaml")
+    forecast_date <- as.Date(model_metadata$forecast_date)
+    filename_suffix <- model_metadata$filename_suffix
+    forecast_months <- model_metadata$forecast_months
+    forecast_years <- model_metadata$forecast_years
+    forecast_newmoons <- model_metadata$forecast_newmoons
+
+  # Forecast All plots
+
+    allresults <- weecology.esss(abundances = all, 
+                                 forecast_date = forecast_date,
+                                 forecast_months = forecast_months, 
+                                 forecast_years = forecast_years,
+                                 forecast_newmoons = forecast_newmoons,
+                                 level = "All",
+                                 num_forecast_months = 12, CI_level = 0.9)
+
+  # Forecast Control plots
+
+    controlsresults <- weecology.esss(abundances = controls, 
+                                      forecast_date = forecast_date,
+                                      forecast_months = forecast_months, 
+                                      forecast_years = forecast_years,
+                                      forecast_newmoons = forecast_newmoons,
+                                      level = "Controls",
+                                      num_forecast_months = 12, 
+                                      CI_level = 0.9)
+
+  # Combine output
+  #  warnings are suppressed here because they're just associated with 
+  #  coercing un-matching sets of factors to character vectors to bind
```

Comment:
My temptation here is to avoid suppressing the warnings by changing the factors into character vectors somewhere. This removes the need to the comment and the (sort of scary) `suppressWarnings` calls.

Reply:
good call. i definitely don't like to use suppressWarnings